pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -124,25 +124,25 @@ func (d *DualWriterMode2) List(ctx context.Context, options *metainternalversion
 	return ll, nil
 }
 
-func enrichObject(orig, copy runtime.Object) (runtime.Object, error) {
-	accessorC, err := meta.Accessor(copy)
-	if err != nil {
-		return nil, err
+// DeleteCollection overrides the behavior of the generic DualWriter and deletes from both LegacyStorage and Storage.
+func (d *DualWriterMode2) DeleteCollection(ctx context.Context, deleteValidation rest.ValidateObjectFunc, options *metav1.DeleteOptions, listOptions *metainternalversion.ListOptions) (runtime.Object, error) {
+	legacy, ok := d.Legacy.(rest.CollectionDeleter)
+	if !ok {
+		return nil, errDualWriterCollectionDeleterMissing
 	}
-	accessorO, err := meta.Accessor(orig)
+
+	// #TODO: figure out how to handle partial deletions
+	deleted, err := legacy.DeleteCollection(ctx, deleteValidation, options, listOptions)
 	if err != nil {
```

Comment:
What do you think about distinguishing between the case where it returns an error because the list was already deleted (not found) and the case when a list could not be deleted because of some other error?
In the first case, I'd say it makes sense to try and delete from storage (because legacy didn't have the list already), whereas in the second one, it makes sense to error out.

Reply:
Good point. This might be more about handling partial deletions than about a "not found" error though. I am not convinced that the list is considered to be a standalone resource in a case like this. Therefore, if there are "not found" errors, they might be referring to an individual entity/list item level rather than the overarching list. 

There's some nuance here. I suggest this be done together with handling partial deletions (added a task for this) and that for now we only consider the happy path.